### DISCUSSION TITLE
Support ipv6 link local addressing

### DISCUSSION THREAD

**Discussion by jbemmel** [ipspace/netlab#90] [Ideas]
I'd like to build a topology where I enable ipv6 on interfaces, without allocating an explicit ipv6 range

The combination should also be possible ( i.e. get both a static ipv6 address AND an auto-generated link-local address )

Perhaps an 'ipv6_link_local' flag (true/false)?

**Comment by ipspace** [comment]
Use https://github.com/ipspace/netsim-tools/blob/master/docs/addressing.md#unnumbered-interface-support ?

Also, most IPv6 implementations generate LLA no matter what, so if an interface has an IPv6 address, it will effectively have configured + LLA address. Am I missing something?

**Reply by jbemmel** [reply]
I did add unnumbered support for SROS, but you cannot enable that and also provision a static address. So for example if you want a static IPv4 address but LLA for ipv6, 'unnumbered' doesn't work

**Reply by ipspace** [reply]
That's why I added "we need per-protocol unnumbered interfaces" as a comment to the other issue. Have to make a separate (long-term) issue out of that, it will be a doozy.

**Comment by ipspace** [comment]
Implemented in #97
